### ISSUE TITLE
Android TV / WebOS / Tizen

### ISSUE THREAD

**Issue by aleksilassila** [aleksilassila/reiverr#127]

**Comment by Marty-SK** [comment]
Hi @aleksilassila,

Im having an older Samsung TV (Tizen 7.0). I was trying to compile a file for my TV, however the script provided didn't worked for me (got some error msg).
Is this a good place where to post error message?

**Comment by aleksilassila** [comment]
@Marty-SK  yeah sure, go ahead. I got Tizen 5.5 so I'm sure we'll get yours running with minimal tweaking

**Comment by Marty-SK** [comment]
Hi @aleksilassila ,

When I run `npm run build:tizen;C:\tizen-studio\tools\ide\bin\tizen.bat build-web -- tizen;C:\tizen-studio\tools\ide\bin\tizen.bat package -t wgt -o .\tizen -- .\tizen\.buildResult\;C:\tizen-studio\tools\ide\bin\tizen.bat install -n .\tizen\Reiverr.wgt -t QE55Q9FNA` in CMD line, Im getting following error:

`npm error Missing script: "build:tizen;C:\tizen-studio\tools\ide\bin\tizen.bat"
npm error
npm error To see a list of scripts, run:
npm error   npm run
npm error A complete log of this run can be found in: C:\Users\Administrator\AppData\Local\npm-cache\_logs\2024-07-09T14_22_33_724Z-debug-0.log`

Im total newbie with npm and compiling the app, so I probably did some mistake in step 1 `Follow the manual installation steps above to install the dependencies (npm install)`

![image](https://github.com/aleksilassila/reiverr/assets/166378138/026be0a9-0625-492e-a763-f3ec1e353b57)

**Comment by aleksilassila** [comment]
@Marty-SK can you make sure you run the command in the source directory for reiverr? That's the directory containing all the reiverr source code as well as a file called "package.json", which is probably what could not be found based on the log

**Comment by Marty-SK** [comment]
@aleksilassila 
steps which I did
1. unpacked reiverr-master to Desktop (C:\Users\Administrator\Desktop\reiverr-master)
2. run CMD line with commands in C:\Users\Administrator\Desktop\reiverr-master
`npm install`
`npm install --prefix backend`
`npm run build`

3. run command 
`C:\Users\Administrator\Desktop\reiverr-master>npm run build:tizen;C:\tizen-studio\tools\ide\bin\tizen.bat build-web -- tizen;C:\tizen-studio\tools\ide\bin\tizen.bat package -t wgt -o .\tizen -- .\tizen\.buildResult\;C:\tizen-studio\tools\ide\bin\tizen.bat install -n .\tizen\Reiverr.wgt -t QE55Q9FNA`

Got error msg
`npm error Missing script: "build:tizen;C:\tizen-studio\tools\ide\bin\tizen.bat"
npm error
npm error To see a list of scripts, run:
npm error   npm run
npm error A complete log of this run can be found in: C:\Users\Administrator\AppData\Local\npm-cache\_logs\2024-07-09T14_56_46_078Z-debug-0.log`

Log file
[2024-07-09T14_56_46_078Z-debug-0.log](https://github.com/user-attachments/files/16145171/2024-07-09T14_56_46_078Z-debug-0.log)

![image](https://github.com/aleksilassila/reiverr/assets/166378138/03445860-2002-4e68-9e2b-e2dff624326b)
![image](https://github.com/aleksilassila/reiverr/assets/166378138/77dacbe7-202a-462f-be34-38005f5f361e)

**Comment by aleksilassila** [comment]
@Marty-SK can you try running the install command in powershell instead of cmd?

**Comment by Marty-SK** [comment]
Thanks @aleksilassila !

CMD line was the culprit. When I run the commmand via Powershell, it works like a charm.

Now, I see the app on my Samsung TV, but when I run the app, I see this:

![image](https://github.com/aleksilassila/reiverr/assets/166378138/d01a0721-dbb5-45da-addd-9793d09a6dee)
Ive tried restart TV, but no luck.

Im runnning latest Reiverr 2.0 on Synology NAS 920 in Docker. Its accessible via web browser without any issues.

![image](https://github.com/aleksilassila/reiverr/assets/166378138/b03f67bb-f03d-48e0-9a87-1cde1aa94bbb)

**Comment by aleksilassila** [comment]
Good to hear that the install script works! I might have to specify the instructions little bit. 

I suspect the error you're seeing now is because of your tizen version, I'll try to investigate and see if I can get it fixed.

**Comment by GastonRAraujo** [comment]
Hi, I'm planning to use this on my samsung tv :)  I found easier to install apps the docker way like the jellyfin app:

1. https://github.com/Georift/install-jellyfin-tizen
It uses a docker image that already have tizen and run the commands in a .sh file

2. https://github.com/babagreensheep/jellyfin-tizen-docker/blob/master/Dockerfile
write everything on the dockerfile

**Comment by aleksilassila** [comment]
> Hi, I'm planning to use this on my samsung tv :)  I found easier to install apps the docker way like the jellyfin app:
> 
> 1. https://github.com/Georift/install-jellyfin-tizen
> It uses a docker image that already have tizen and run the commands in a .sh file
> 
> 2. https://github.com/babagreensheep/jellyfin-tizen-docker/blob/master/Dockerfile
> write everything on the dockerfile

Woah I didn't realize that's possible, I should do similar docker image for installing Reiverr

**Comment by GastonRAraujo** [comment]
I will try do something similar with my tv and post it here if it works

**Comment by aleksilassila** [comment]
Hey @GastonRAraujo, did you get it to work? I added a docker-based installation method [here](https://github.com/aleksilassila/reiverr/tree/7165673b18b28f1712b01b0e72cc9905c3de8e58?tab=readme-ov-file#tizen--samsung-smart-tvs), you can try it out if you like. If you do, let me know how it goes

**Comment by GastonRAraujo** [comment]
After that night, a heavy rain flooded my city and I lost power for a week, so I didn't test my idea. However, I took the dev branch and installed it on my TV using the following commands:

1. **Download**  
   Download the zip file or clone the repo.

2. **Update File Permissions**  
   From the root directory, update the file permissions for the shell script:
   ```bash
   chmod +x tizen/install.sh
   ```

3. **Build**  
   Build the Docker image:
   ```bash
   docker build -f tizen/Dockerfile -t my-tizen-app .
   ```

4. **Run**  
   Run the Docker container:
   ```bash
   docker run --rm my-tizen-app $TV_IP
   ```
   *Example:*
   ```bash
   docker run --rm my-tizen-app 192.168.1.100
   ```

The installation ran successfully. I will update you after some usage. Thank you for your work 😄

**Comment by GastonRAraujo** [comment]
The installation was successfully but Im stuck at the setup... Im using a raspberry pi and a tizen tv. On the raspberry Im running jellyfin, *arrs apps and also clone the repo and build from source the backend folder. Got an error about plugings missings, create a new folder named plugins and then run
`npm run build`
`npm start:prod`

But I cannot found a way to connect to Sonnar or Raddar. On integrations only have Tmdb

**Comment by aleksilassila** [comment]
Hey @GastonRAraujo, glad the docker installation worked! Was the chmod part necessary to run the image? If so, I can add the command to the Dockerfile to automate that step. 

As for the sonarr and radarr integrations, they were replaced in favor of the new plugin system that allows streaming from various sources. I might add a plugin for them later though. You can see more information [here](https://github.com/aleksilassila/reiverr/releases/tag/v2.1.1)

Stay safe!

**Comment by GastonRAraujo** [comment]
The chmod is mandatory, you need to include it any time you run a bash file. If can put it in the dockerfile it would be grate.

If you can include the request from v1 as a plugin it would be grate, thats the reason I decided to use this instead of jellyfin :)

Im having an issue that the library doesnt show anything but in jellyfin I have several movies and series. Do you know what could be the issue?
Also there is a modal that pop up about some connection after login that is buggy. The remote cannot control the buttons, instead it control the app in the background. I need to open something and then if I click back it closes the window.

As for the setup I think more details/tutorials are needed to make it easier for the users.

Let me know if you need any help and if you know what is wrong with my jellyfin connection not showing my library

**Comment by trumblejoe** [comment]
Can we get FIRE TV APK's added to releases for side-loading?

**Comment by aleksilassila** [comment]
> The chmod is mandatory, you need to include it any time you run a bash file. If can put it in the dockerfile it would be grate.

Done!

> Im having an issue that the library doesnt show anything but in jellyfin I have several movies and series. Do you know what could be the issue? 

In the latest release, the watch history and library page is separate from Jellyfin and maintained by Reiverr alone. Currently the library page only shows titles that you have added to your library manually from within Reiverr (Add to library button). To improve this, in next release the library page will show separately "my list" items (the manually added items), and then additionally it will have separate pages for catalogues provided by plugins. So you will be able to browse all the items that you have in your Jellyfin library from within Reiverr if you have connected through the Jellyfin plugin.

> Also there is a modal that pop up about some connection after login that is buggy. The remote cannot control the buttons, instead it control the app in the background. I need to open something and then if I click back it closes the window.

I will look into this

> As for the setup I think more details/tutorials are needed to make it easier for the users.

Agreed, I will try to improve on this in the future releases

> Can we get FIRE TV APK's added to releases for side-loading?

Yes, the plan is to have android tv / fireos builds hopefully in next release or one after

**Comment by Aerya** [comment]
Hello, any update on this?
